### ISSUE TITLE
finetune the model for extrating feature

### ISSUE THREAD

**Issue by BrettLL** [facebook/fb.resnet.torch#26]
I want to use my own data to finetune the model for extrating feature, but the finetune procession  just change the Linner layer(in file init.lua), I want to chang more layers backward, how can i do?? looking forward to your reply~

**Comment by colesbury** [comment]
The [fine-tuning example](https://github.com/facebook/fb.resnet.torch/tree/master/pretrained#fine-tuning-on-a-custom-dataset) trains the entire network. The `-replaceClassifier` option in [models/init.lua](https://github.com/facebook/fb.resnet.torch/blob/master/models/init.lua#L63-L75) will replace the Linear layer, but gradients are propagated through the entire network.

**Comment by BrettLL** [comment]
Thank you, I see. However, if I want to fine-tune from special layer A, and the parameters before layer A would not be changed again. As in caffe prototxt, I can set the param lr_multi and decay_multi are zero. Can I do like that?? Thanks~~

**Comment by colesbury** [comment]
I'm not very familiar with Caffe, so I don't know what lr_multi/decay_multi are. If you want to *only* train the Linear layer then you need to modify that code so that it doesn't train the base of the network.

**Comment by ThakurRajAnand** [comment]
@colesbury How can we predict on 100K new images using the fine-tuned model and save the predictions into CSV.

Thanks in advance.
